packages/CatalogService/src/apiFactory/operations/searchFactory.ts

```diff
@@ -1,90 +1,72 @@
 import { z } from "zod";
-import { RDFTripleSchema } from "@telicent-oss/rdfservice";
-import { CatalogService, formatDataAsArray } from "../../../index";
+import { CatalogService, DCATResource } from "../../../index";
 
-import {
-  UIDataResourceSchema,
-  uiDataResourceFromInstanceWithTriples,
-  typeStatementMatcherWithId,
-  DCATResourceSchema,
-  UISearchParamsType,
-  getAllRDFTriples,
-  UIDataResourceType,
-} from "./utils/common";
-import { getAllResourceTriples } from "./utils/getAllResourceTriples";
+import { UIDataResourceSchema, UISearchParamsType } from "./utils/common";
 import { transformDataResourceFilters } from "./utils/transformDataResourceFilters";
-import { tryCatch } from "./utils/tryCatch";
-import { session } from "../../constants";
-import { tryInstantiate } from "./utils/tryInstantiate/tryInstantiate";
 
 export const searchFactory = (service: CatalogService) => {
+  let cachedResources: DCATResource[] | undefined = undefined;
+  const getAllResources = async () => {
+    if (!cachedResources) {
```

Comment:
Non-blocking: Any thoughts around conditions for invalidating the cache?

Also, as you know, the services do have some caching functionality. I assume that was not appropriate here?